src/merkle_tree.rs

```diff
@@ -0,0 +1,206 @@
+use hmac_sha256::Hash;
+
+pub struct MerkleTree {
+    pub root: Option<[u8; 32]>,
+    pub leaves: Vec<[u8; 32]>,
+}
```

Comment:
Is there any reason you chose not to store the intermediate branches? If so, the rationale and tradeoffs made should be documented either here or in the README, or both.

Reply:
If I understand correctly, the leaves (given the same ordering) are enough to compute the Merkle Tree. Storing the intermediate nodes is more of an optimization to avoid recomputing the tree each time it is queried. I don't rule out storing the intermediate nodes in the future, I just don't think it is necessary right now.

Do you agree?